Pidaq/Lib/uploadToDrive/driveClass.py

```diff
@@ -0,0 +1,127 @@
+    
+            with open('drive_options.json') as json_file:
+                self.parentDictionary = json.load(json_file)
+
+
+    def execute(self, filePath, parentName):
+        """ Uploads a csv file from a local directory to a folder on Google Drive.
+        Parameters:
+        "filePath" is the path of the csv file itself and
+                    should be passed without quotations.
+        "parentName" is the name of the testing folder in the drive
+                and only a specific set of values are allowed.
+        The parent folder is supposedly going to be fixed for a
+        long period of time.
+        """
+                # Searches for the parent name in the dictionary
+        lowerDictionary = {}
+        for key, value in self.parentDictionary.items():
+            lowerDictionary[key.lower()] = value
+        try:
+            self.parentFolderID = lowerDictionary[parentName.lower()]
+        except NameError:
+            # Uploads to a default folder if the folder name does
+            # Not exist in the dictionary
+            self.parentFolderID = self.parentDictionary['default']
+            logging.error('Folder name not found \nuploading to "Default"')
+                
+        # Today's date in YYYY-MM-DD format is stored in the variable "date"
+        date = datetime.today().strftime('%Y-%m-%d')
+        
+        # If the folder does not exist, should be the case
+        # When running for the first time on a given day
+        if self.checkExistingSubFolder(date) == 0:
+            folder_metadata = {
+                'name' : date,
+                'mimeType': 'application/vnd.google-apps.folder',
+                'parents' : [self.parentFolderID]
+                }
+            # Creates a folder with the given date
+            folder = self.DRIVE.files().create(body = folder_metadata,
+                                               fields = 'id').execute()
+            # Returns the date folder's ID
+            folderId = self.getFolderId(date)
+        # Executes if the folder already exists  
+        elif self.checkExistingSubFolder(date) == 1:
+            logging.warning("found folder for today's date")
+            folderId = self.getFolderId(date)
+        # Executes if more than one folder exists
+        else:
+            logging.warning('unexpected count of folders named with a given date')
+            return
+        
+        # Extracts the .csv file name from the path
+        filename= os.path.basename(filePath)
+        metadata = {'name': filename, 'parents' : [folderId] }
+        # Uploads the .csv file
+        response = self.DRIVE.files().create(body=metadata,
+                                             media_body=filePath).execute()
+        
+        if response:
+            logging.warning('Uploaded "%s" (%s)' % (metadata['name'], response['mimeType']))
+        else:
+            logging.warning('error in uploading the csv file')

```

Comment:
Other way around for this, I feel like this might be better suited to be logging.error